applications/functions/src/functions_call.erl

```diff
@@ -0,0 +1,631 @@
+%% build coudhDB show function http url
+build_voice_uri(Host, DocId) ->
+    HttpHost = http_host(Host),
+    iolist_to_binary([HttpHost, ?FUNC_PATH, ?DEFAULT_FUNC_EVAL, DocId]).
+
+-spec http_host(binary() | list()) -> kz_term:ne_binary().
+http_host(Host) ->
+    %% https also starts with http
+    %% https has to be explicitly specified by user.
+    case lists:prefix("http", kz_term:to_list(Host)) of
+        true -> Host;
+        false -> iolist_to_binary([<<"http://">>, Host])
+    end.
+
+-spec send_req(kapps_call:call(), kz_term:ne_binary(), 'post', kz_json:object() | kz_term:proplist(), kz_term:ne_binary(), boolean()) ->
+                      {'ok', kz_http:req_id(), kapps_call:call()} |
+                      {'stop', kapps_call:call()}.
+send_req(Call, Uri, Method, BaseParams, ReqBodyFormat, Debug) when not is_list(BaseParams) ->
+    send_req(Call, Uri, Method, kz_json:to_proplist(BaseParams), ReqBodyFormat, Debug);
+send_req(Call, Uri, 'post', BaseParams, ReqBodyFormat, Debug) ->
+    UserParams = kzt_translator:get_user_vars(Call),
+    Params = kz_json:set_values(BaseParams, UserParams),
+    UpdatedCall = kapps_call:kvs_erase(<<"digits_collected">>, Call),
+    Headers = [{"Content-Type", req_content_type(ReqBodyFormat)}],
+    send(UpdatedCall, Uri, Headers, format_request(Params, ReqBodyFormat), Debug).
+
+-spec send(kapps_call:call(), kz_term:ne_binary(), kz_term:proplist(), iolist(), boolean()) ->
+                  {'ok', kz_http:req_id(), kapps_call:call()} |
+                  {'stop', kapps_call:call()}.
+send(Call, Uri, ReqHdrs, ReqBody, Debug) ->
+    lager:debug("sending req to ~s: ~s", [Uri, iolist_to_binary(ReqBody)]),
+
+    maybe_debug_req(Call, Uri, 'post', ReqHdrs, ReqBody, Debug),
+
+    case kz_http:async_req(self(), 'post', Uri, ReqHdrs, ReqBody) of
+        {'http_req_id', ReqId} ->
+            lager:debug("response coming in asynchronously to ~p", [ReqId]),
+            {'ok', ReqId, Call};
+        {'error', _Reason} ->
+            lager:debug("error with req: ~p", [_Reason]),
+            {'stop', Call}
+    end.
+
+-spec normalize_resp_headers(kz_term:proplist()) -> kz_term:proplist().
+normalize_resp_headers(Headers) ->
+    [{kz_term:to_lower_binary(K), kz_term:to_binary(V)} || {K, V} <- Headers].
+
+-spec handle_resp(kz_term:api_binary(), kapps_call:call(), kz_term:ne_binary(), binary(), pid()) -> 'ok'.
+handle_resp(RequesterQ, Call, CT, <<_/binary>> = RespBody, AMQPConsumer) ->
+    _ = kz_amqp_channel:consumer_pid(AMQPConsumer),
+
+    kz_util:put_callid(kapps_call:call_id(Call)),
+    Srv = kzt_util:get_amqp_listener(Call),
+
+    case process_resp(RequesterQ, Call, CT, RespBody) of
+        {'stop', Call1} -> stop_call(Srv, Call1);
+        {'ok', Call1} -> stop_call(Srv, Call1);
+        {'usurp', _Call1} -> usurp_executor(Srv);
+        {'request', Call1} ->
+            updated_call(Srv, kzt_util:increment_iteration(Call1)),
+            new_request(Srv
```

Comment:
spec for new_request/3 is `-spec new_request(pid(), kz_term:ne_binary(), kz_json:object()) -> 'ok'.` but kzt_util:get_voice_uri_method/1 will pass in 'get' | 'post'. Are you sure this is what you want?